akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -75,20 +75,20 @@ abstract class GraphStageLogic {
   /**
    * INTERNAL API
    */
-  private[stream] var inHandlers = scala.collection.Map.empty[Inlet[_], InHandler]
+  private[stream] var inHandlers = Array.ofDim[InHandler](16) // FIXME: proper size
   /**
    * INTERNAL API
    */
-  private[stream] var outHandlers = scala.collection.Map.empty[Outlet[_], OutHandler]
+  private[stream] var outHandlers = Array.ofDim[OutHandler](16)
```

Comment:
these should be sized accordingly to the Shape of course. This was just a quick hack to get this working.